src/test/java/org/jenkinsci/plugins/github_branch_source/GithubAppCredentialsAppInstallationTokenTest.java

```diff
@@ -0,0 +1,69 @@
+package org.jenkinsci.plugins.github_branch_source;
+
+import org.junit.Test;
+
+import java.time.Duration;
+import java.time.Instant;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+
+public class GithubAppCredentialsAppInstallationTokenTest {
+
+    @Test
+    public void testAppInstallationTokenStale() throws Exception {
+
+        GitHubAppCredentials.AppInstallationToken token;
+        long now;
+
+        now = Instant.now().getEpochSecond();
+        token = new GitHubAppCredentials.AppInstallationToken("", now);
+        assertThat(token.isStale(), is(false));
+        assertThat(token.getTokenStaleEpochSeconds(), equalTo(now + GitHubAppCredentials.AppInstallationToken.NOT_STALE_MINIMUM_SECONDS));
```

Comment:
Observed to flake in #527:

```
java.lang.AssertionError: 

Expected: <1648072806L>
     but: was <1648072807L>
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:6)
	at org.jenkinsci.plugins.github_branch_source.GithubAppCredentialsAppInstallationTokenTest.testAppInstallationTokenStale(GithubAppCredentialsAppInstallationTokenTest.java:23)
```

I do not think you can make an assertion like this without some grace period.